src/pip/_internal/commands/configuration.py

```diff
@@ -51,14 +50,8 @@ class ConfigurationCommand(Command):
         %prog [<file-option>] unset name
     """
 
-    def __init__(self, name, summary, isolated=False):
-        super(ConfigurationCommand, self).__init__(
-            name, summary, isolated=isolated
-        )
-
-        self.configuration = None
```

Comment:
If we remove the `__init__` definition from here, where would we initialize the `self.configuration`? IMO, it shouldn't be removed. While working on #8281, I intentionally kept the `__init__` definition, due to the variable initialization.

Reply:
We will initialize it in the `run` method.

https://github.com/pypa/pip/blob/d01bfcfaa13a4f06fa0ce61fa18cf06012f2e78f/src/pip/_internal/commands/configuration.py#L128-L131

And I also see that the first function which is called  after updating the `options` in [base_command._main](https://github.com/pypa/pip/blob/master/src/pip/_internal/cli/base_command.py#L124) is that `run` method, `self.configuration` will always be initialized before any of it's functions are called.

Reply:
@deveshks is there a particular reason why you need to remove the constructor? I personally like to have instance variables initialized in `__init__.py`, as a way to easily make them discoverable.

Reply:
@sbidoul , as I mentioned in https://github.com/pypa/pip/pull/8365#discussion_r432891959, `self.configuration` was not actually being initialized there, but in the `run` method, so I thought that `__init__` is not really doing anything here, so I removed it.

I can readd it if needed, but then a lot of places where `self.configuration` is being used, I would have to add `assert self.configuration` since instantiating `self.configuration = None` in `__init__` makes it `Optional`. Do you have any suggestions to avoid it?

Reply:
Ok I see. In a way I wonder if doing this is not relying on a mypy loophole. Indeed mypy has no guarantee that run() will have been called before the other methods, so it could as well complain that the configuration attribute is not guaranteed to exist. The proper way would probably be to pass the configuration object around to all methods that need it instead of having it as an attribute.

It's not an issue in practice because *we* know that run will always be called first, so :+1:

Reply:
So I assume from your response that I can keep this as is, and no change is needed here?